phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolIT.java

```diff
@@ -81,681 +86,850 @@
-        int index = 1;
-        // insert row
-        stmt.setInt(index++, id);
-        stmt.setString(index++, name);
-        stmt.setInt(index++, zip);
-        stmt.setTimestamp(index++, new Timestamp(testTime));
-        stmt.executeUpdate();
-    }
+        @After public void teardown() throws SQLException {
+            if (connGlobal != null) {
+                connGlobal.close();
+            }
+            if (connTenant != null) {
+                connTenant.close();
+            }
+        }
 
-    private int deleteRow(String fullTableName, String whereCondition) throws SQLException {
-        String deleteSql = String.format(DELETE_SQL, indexTableFullName) + whereCondition;
-        PreparedStatement deleteStmt = conn.prepareStatement(deleteSql);
-        return deleteStmt.executeUpdate();
-    }
+        /**
+         * Tests that the config for max number of output rows is observed
+         */
+        @Test public void testTenantViewAndIndexEqual() throws Exception {
+            connTenant.createStatement()
+                    .execute(String.format(upsertQueryStr, tenantViewName, tenantId, 1, "x"));
+            connTenant.commit();
+
+            String[] argValues =
+                    getArgValues("", tenantViewName, indexNameTenant, 10L,
+                            SourceTable.INDEX_TABLE_SOURCE, false, null, null, tenantId,
+                            EnvironmentEdgeManager.currentTimeMillis());
+
+            List<Job> completedJobs = runScrutiny(argValues);
+            // Sunny case, both index and view are equal. 1 row
+            assertEquals(1, completedJobs.size());
+            for (Job job : completedJobs) {
+                assertTrue(job.isSuccessful());
+                Counters counters = job.getCounters();
+                assertEquals(1, getCounterValue(counters, VALID_ROW_COUNT));
+                assertEquals(0, getCounterValue(counters, INVALID_ROW_COUNT));
+            }
+        }
 
+        /**
+         * Tests global view on multi-tenant table should work too
+        **/
+        @Test public void testGlobalViewOnMultiTenantTable() throws Exception {
+            String globalViewName = generateUniqueName();
+            String indexNameGlobal = generateUniqueName();
+
+            connGlobal.createStatement().execute(
+                    String.format(createViewStr, globalViewName, multiTenantTable));
+
+            String idxStmtGlobal = String.format(createIndexStr, indexNameGlobal, globalViewName);
+            connGlobal.createStatement().execute(idxStmtGlobal);
+            connGlobal.createStatement()
+                    .execute(String.format(upsertQueryStr, globalViewName, "global", 5, "x"));
+            connGlobal.commit();
+            String[] argValues =
+                    getArgValues("", globalViewName, indexNameGlobal, 10L,
+                            SourceTable.INDEX_TABLE_SOURCE, false, null, null, null,
+                            En
```

Comment:
@gokceni - could you please file a followup JIRA to allow for output to table when using a tenant id connection.

Reply:
Done https://issues.apache.org/jira/browse/PHOENIX-5168